cfgov/v1/documents.py

```diff
@@ -0,0 +1,201 @@
+from django_elasticsearch_dsl import Document, fields
+from django_elasticsearch_dsl.registries import registry
+
+from v1.models.base import (
+    CFGOVAuthoredPages, CFGOVPageCategory, CFGOVTaggedPages
+)
+from v1.models.learn_page import AbstractFilterPage
+
+
+@registry.register_document
+class FilterablePagesDocument(Document):
+
+    tags = fields.ObjectField(properties={
+        'slug': fields.KeywordField()
+    })
+    categories = fields.ObjectField(properties={
+        'name': fields.KeywordField()
+    })
+    authors = fields.ObjectField(properties={
+        'name': fields.TextField(),
+        'slug': fields.KeywordField()
+    })
+    title = fields.TextField(attr='title')
+    is_archived = fields.KeywordField(attr='is_archived')
+    content = fields.TextField()
+    date_published = fields.DateField(attr='date_published')
+    url = fields.KeywordField()
+    start_dt = fields.DateField()
+    end_dt = fields.DateField()
+    statuses = fields.KeywordField()
+    initial_filing_date = fields.DateField()
+
+    def get_queryset(self):
+        return AbstractFilterPage.objects.live().public()
+
+    def prepare_content(self, instance):
+        try:
+            content_field = instance._meta.get_field('content')
+            value = content_field.value_from_object(instance)
+            content = content_field.get_searchable_content(value)
+            content = content.pop()
+            return content
+        except Exception:
```

Comment:
[PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) recommends catching "specific exceptions whenever possible". What is being caught here? Can `Exception` be more specific?

Reply:
Yeah I meant to force this exception to catch the specific type - it occurs on the case the specific Page Model doesn't have the field being referenced.